shogun/lib/external/ssl.cpp

```diff
@@ -139,7 +139,7 @@ int32_t CGLS(
 		for (i=0; i < active; i++)
 		{
 			ii=J[i];
-			t=features->dense_dot(ii, p, n-1);
+			t=features->dot(ii, p.slice(0, n-1));
```

Comment:
Is `slice` needed here, or is it using the complete vector effectively?

Reply:
It's needed, `p` is of size `n`, so the last element is discarded, otherwise dot will throw an error.